docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,489 @@
+The Runtime additionally provides the ``<<< >>>`` syntax for launching kernels, which resembles a special function call and is easier to use than explicit launch API, in particular with respect to handling of kernel arguments.
+However, this syntax is not standard C++ and is available only when NVCC is used to compile the host code.
+
+Additional Information
+^^^^^^^^^^^^^^^^^^^^^^
+
+HIP-Clang creates a primary context when the HIP API is called. So in a pure driver API code, HIP-Clang will create a primary context while HIP/NVCC will have an empty context stack. HIP-Clang will push primary context to the context stack when it is empty. This can have subtle differences on applications which mix the runtime and driver APIs.
+
+``hip-clang`` Implementation Notes
+----------------------------------
+ 
+``.hip_fatbin``
+^^^^^^^^^^^^^^^
+
+hip-clang links device code from different translation units together. For each device target, a code object is generated. Code objects for different device targets are bundled by ``clang-offload-bundler`` as one fatbinary, which is embedded as a global symbol ``__hip_fatbin`` in the ``.hip_fatbin`` section of the ELF file of the executable or shared object.
+
+Initialization and Termination Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+hip-clang generates initialization and termination functions for each translation unit for host code compilation. The initialization functions call ``__hipRegisterFatBinary`` to register the fatbinary embedded in the ELF file. They also call ``__hipRegisterFunction`` and ``__hipRegisterVar`` to register kernel functions and device side global variables. The termination functions call ``__hipUnregisterFatBinary``.
+hip-clang emits a global variable ``__hip_gpubin_handle`` of ``void**`` type with ``linkonce`` linkage and initial value 0 for each host translation unit. Each initialization function checks ``__hip_gpubin_handle`` and register the fatbinary only if ``__hip_gpubin_handle`` is 0 and saves the return value of ``__hip_gpubin_handle`` to ``__hip_gpubin_handle``. This is to guarantee that the fatbinary is only registered once. Similar check is done in the termination functions.
+
+Kernel Launching
+^^^^^^^^^^^^^^^^
+
+hip-clang supports kernel launching via the CUDA ``<<<>>>`` syntax, ``hipLaunchKernel`` or ``hipLaunchKernelGGL``. The latter one is a macro which expands to the CUDA ``<<<>>>`` syntax by default, or can be turned into a template by defining ``HIP_TEMPLATE_KERNEL_LAUNCH``.
+
+When the executable or shared library is loaded by the dynamic linker, the initialization functions are called. In the initialization functions, when ``__hipRegisterFatBinary`` is called, the code objects containing all kernels are loaded; when ``__hipRegisterFunction`` is called, the stub functions are associated with the corresponding kernels in code objects.
+
+hip-clang implements two sets of kernel launching APIs.
+
```

Comment:
Probably just delete this line to make the single sentence above part of the paragraph below ( "HIP-Clang implements two sets of APIs for launching kernels" should not be a single-sentence paragraph).

Reply:
Yes, you are right. Updated.